### PR TITLE
refactor(editor): Migrate FixedCollectionParameter to composition API

### PR DESCRIPTION
**Pull request by ShireenMissi**
## Summary
This PR migrates FixedCollectionParameter to composition API

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/components/FixedCollectionParameter.vue [hunk 1]
```diff
@@ -1,220 +1,201 @@
...
-				this.mutableValues = deepCopy(newValues);
-			},
-			deep: true,
-		},
-	},
-	created() {
-		this.mutableValues = deepCopy(this.values);
+const locale = useI18n();
+
+const props = defineProps<{
```
Comment (elsmr) [inline packages/editor-ui/src/components/FixedCollectionParameter.vue:14]: use `withDefaults` here to set the same default values as before. Just to be sure the component behaves exactly the same as before.

File: packages/editor-ui/src/components/FixedCollectionParameter.vue [hunk 2]
```diff
@@ -1,220 +1,201 @@
...
+	nodeValues: INodeParameters;
+	parameter: INodeProperties;
+	path: string;
+	values: Record<string, INodeParameters[]>;
+	isReadOnly: boolean | undefined;
```
Comment (elsmr) [inline packages/editor-ui/src/components/FixedCollectionParameter.vue:19]: values and isReadOnly should be optional (they don't have required=true in defineComponent)

File: packages/editor-ui/src/components/FixedCollectionParameter.vue [hunk 3]
```diff
@@ -1,220 +1,201 @@
...
+			returnProperties.push(tempProperties);
+		}
+	}
+	return returnProperties;
+});
+const multipleValues = computed(() => {
+	return !!props.parameter.typeOptions?.multipleValues;
+});
+const parameterOptions = computed(() => {
+	if (multipleValues && isINodePropertyCollectionList(props.parameter.options)) {
```
Comment (elsmr) [inline packages/editor-ui/src/components/FixedCollectionParameter.vue:50]: multipleValues.value

File: packages/editor-ui/src/components/FixedCollectionParameter.vue [hunk 4]
```diff
@@ -1,220 +1,201 @@
...
+const props = defineProps<{
+	nodeValues: INodeParameters;
+	parameter: INodeProperties;
+	path: string;
+	values: Record<string, INodeParameters[]>;
+	isReadOnly: boolean | undefined;
+}>();
+
+const emit = defineEmits<{
+	valueChanged: [value?: any];
```
Comment (elsmr) [inline packages/editor-ui/src/components/FixedCollectionParameter.vue:23]: Specify a type for `value`. From usage I think it should be something like
```ts
type ValueChangedEvent = { path: string; value: NodeParameterValueType; type?: 'optionsOrderChanged'; }
```

File: packages/editor-ui/src/components/FixedCollectionParameter.vue [hunk 5]
```diff
@@ -1,220 +1,201 @@
...
+			// Add a new option
+			newParameterValue[optionParameter.name] = deepCopy(optionParameter.default);
+		}
+	}
 
-				newValue.push(newParameterValue);
-			} else {
-				newValue = newParameterValue;
-			}
+	let newValue;
```
Comment (elsmr) [inline packages/editor-ui/src/components/FixedCollectionParameter.vue:172]: If possible set a type for newValue, maybe `NodeParameterValueType`

File: packages/editor-ui/src/components/FixedCollectionParameter.vue [hunk 6]
```diff
@@ -1,220 +1,201 @@
...
+	if (multipleValues) {
```
Comment (elsmr) [inline packages/editor-ui/src/components/FixedCollectionParameter.vue:173]: multipleValues.value

File: packages/editor-ui/src/components/FixedCollectionParameter.vue [hunk 7]
```diff
@@ -1,220 +1,201 @@
...
+});
+const multipleValues = computed(() => {
+	return !!props.parameter.typeOptions?.multipleValues;
+});
+const parameterOptions = computed(() => {
+	if (multipleValues && isINodePropertyCollectionList(props.parameter.options)) {
+		return props.parameter.options;
+	}
+
+	return (props.parameter.options as INodePropertyCollection[]).filter((option) => {
```
Comment (elsmr) [inline packages/editor-ui/src/components/FixedCollectionParameter.vue:54]: I think this type cast is avoidable by doing `props.parameter.options ?? []`, would be safer too 😄

**Review by elsmr — CHANGES_REQUESTED** [review]

File: packages/editor-ui/src/components/FixedCollectionParameter.test.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+describe('FixedCollectionParameter.vue', () => {
+	const pinia = createTestingPinia({
+		initialState: {
+			[STORES.SETTINGS]: {
+				settings: SETTINGS_STORE_DEFAULT_STATE.settings,
+			},
+		},
+	});
+	setActivePinia(pinia);
+	let props: any;
```
Comment (ShireenMissi) [inline packages/editor-ui/src/components/FixedCollectionParameter.test.ts:17]:
```suggestion
	let props = {};
```

**Review by elsmr — APPROVED** [review]

**Pull request merged**